<!-- -->

<launch>		<!-- small ones have land height -0.25, mass 0.8 and emergency thrust 60, bigger ones -0.37, 1.1 and 70 -->
	<arg name="qcID" value="8" /> 
	<arg name="firmware" value="3263" /> <!--old version: 1641, smurf: 3263, new old firware for ID7: 3395-->
	<arg name="rotortype" value="Quadcopter" />
	<arg name="rawAccOffsetX" value="507" />
	<arg name="rawAccOffsetY" value="494" />
	<arg name="takeOffHeight" value="-0.0" />
	<arg name="landHeight" value="-0.25" />
	<arg name="initialMass" value="0.8" />
	<arg name="emergencyThrust" value="60" />
	

	<!-- Vicon -->
	<node name="Vicon" pkg="telekyb_vicon" type="telekyb_vicon">
	      <param name="ViconTracker/tTrackerHostname" value="192.168.111.100:801" />
	</node>
	
	<!-- Joystick Logitech 510 Remapping -->
	<node name="Joystick_$(arg qcID)" pkg="telekyb_joystick" type="tJoy">
	      <param name="JoystickOptions/tDevicePath" value="/dev/input/js0" />
	      <param name="JoystickOptions/tPubName" value="joy_exp" />
		  <param name="JoystickOptions/tButtonRemapping" value="[4,1,2,3,0,5,7,8,6]" />
	      <param name="JoystickOptions/tAxesRemapping" value="[5,2,3,1,0,4]" />
	      <param name="JoystickOptions/tAxisMultiplier" value="[-0.5,0.5,-0.5,0.5,0.5]" /> <!--"[-1.0,1.0,-1.0,1.0,1.0]" /-->
	</node>
	
	
	
	<node name="TeleKybCore_$(arg qcID)" pkg="telekyb_core" type="telekyb_core">
	      <param name="TeleKybCore/tRobotID" value="$(arg qcID)" />
	      <param name="StateEstimatorController/tPluginLookupName" value="tk_state/ViconStateEstimator"/>
	      <param name="StateEstimatorController/tPublishRosTransform" value="false"/>
	      <param name="ViconStateEstimator/tViconSeTopicName" value="/TeleKyb/Vicon/$(arg rotortype)_$(arg qcID)/$(arg rotortype)_$(arg qcID)" />
	      <param name="StandardTrajectoryTracker/tCommandsTopic" value="/TeleKyb/$(arg qcID)/commands" />
	      <param name="PositionControl/tDerGainVelMode" value="12.5" />
	      <param name="TrajectoryProcessorController/tTrajectoryModules" value='[]'/>
	      <param name="StandardMassEstim/tInitialMass" value="0.70"/>
          <param name="tCenterValueX" value="512" />
          <param name="tCenterValueY" value="512" />
		  <param name="StandardMassEstim/tInitialMass" value="$(arg initialMass)"/>
	      <param name="tTakeOffDestination" value="[0.0, 0.0, $(arg takeOffHeight)]"/>
	      <param name="tLandDestinationHeight" value="$(arg landHeight)" />
	</node>

	<!-- MK -->
	<node name="MKInterface_$(arg qcID)" pkg="tk_mkinterface" type="mkinterface">
	      <param name="MKInterface/tUavId" value="$(arg qcID)" />
	      <param name="MKInterface/tUavFirmware" value="$(arg firmware)" />
	      <param name="MKInterface/tInitialDriftEstim" value="true" />
	      <param name="MKInterfaceConnection/tBaudRate" value="BAUD115200" />
	      <param name="MKROSInterface/tCommandsTopic" value="/TeleKyb/$(arg qcID)/commands" />
	      <param name="MKROSInterface/tEmergencyLandService" value="/TeleKyb/TeleKybCore_$(arg qcID)/Behavior/EmergencyLand"/>
		  <param name="MKInterface/tEmergencyThrust" value="$(arg emergencyThrust)" />
	</node>
	
	


	<!-- Obstacle Provider -->
	<node name="Obstacle_$(arg qcID)" pkg="tk_obstacle" type="obstacle_provider">
		<param name="ObstacleProviderController/tObstacleProviders" value='[ "tk_obstacle/SurroundingBox" ]'/>
		<param name="ObstacleProviderController/tTKStateTopicName" value="/TeleKyb/TeleKybCore_$(arg qcID)/Sensor/TKState" />
		<!-- TrackingHall -->
		<param name="SurroundingBox/tMinSurrObsBox" value='[-2.5, -3.5, -3]'/>
		<param name="SurroundingBox/tMaxSurrObsBox" value='[2.5, 3, 0.25]'/>
		<!--param name="ExternalFace/tInputFaces" value="[ [3.05, 0.0, 0.0], [0.0,1.9,0.0], [0.0,0.0,-2.0]]" /-->
	</node>	



	<!-- Experiment -->
	<node name="tk_exp_RCjoy" pkg="tk_exp_RCjoy" type="tk_exp_RCjoy">
	      <param name="ExperimentOptions/robotID" value="$(arg qcID)" />
	      <param name="ExperimentOptions/tJoystickTopic" value="/TeleKyb/tJoy/joy_exp" />
	      <param name="ExperimentOptions/tUseMKInterface" value="true" />
	</node>

</launch>
